client.cpp

```diff
@@ -256,8 +257,14 @@ void client::create_arbitrary_request(const arbitrary_command* cmd, struct timev
 
             assert(key != NULL);
             assert(key_len > 0);
-
-            cmd_size += m_connections[conn_id]->send_arbitrary_command(arg, key, key_len);
+            //when we have static data mixed with the key placeholder
+            if (arg->data.length() != strlen(KEY_PLACEHOLDER)) {
+                std::string str (arg->data);
```

Comment:
I don't think this is good idea to do this str.replcae in "run time", on every request that we send.
I think we can preapre the prefix and suffix of the __key__ before and then just send them before and after. what do you think?

Reply:
@YaacovHazan I've followed your review and we're now using recalculated prefix and suffix strings ( only when required ).

Reply:
Now that key_type is composed by prefix + key + suffix what do you think about something like this:

```
if (prefix>0)
   send_arbitrary_command(prefix)

send_arbitrary_command(key)

if (suffix>0)
   send_arbitrary_command(suffix)
```

Reply:
@YaacovHazan  the approach of:
```
if (prefix>0)
   send_arbitrary_command(prefix)
send_arbitrary_command(key)
if (suffix>0)
   send_arbitrary_command(suffix)
would make us send 3 bulk strings ( when we should send 1 )
```
 believe it's not feasible to be done that way
notice the `$%d\r\n` sent each time we call send_arbitrary_command